### DISCUSSION TITLE
Extension development how to extend nav_entries UI

### DISCUSSION THREAD

**Discussion by 9pr19** [FreshRSS/FreshRSS#5736] [Q&A] [answered]
Hello,

I would like to play around a bit and try to add some buttons to nav_entries https://github.com/FreshRSS/FreshRSS/blob/edge/app/layout/nav_entries.phtml . Unfortunately it seems there is no standard way to do it:
* registerViews does not seems to be of any use since nav_entries is in layout and not in views
* there is no available hook to manage that part of the UI

Does anybody have an idea?

My goal is to add the following buttons to the nav_entries:
* toggle read
* toggle favorite
* why not a quick label button (if it remains enough space)

Thank you :)

**Comment by Alkarex** [comment]
Not exactly the same, but have you considered `$this->registerHook('nav_menu', ...)` ?

**Reply by 9pr19** [reply]
@Alkarex I’ve tried the code below but it adds an item to the nav_menu and not to the nav_entries buttons:

```php
<?php
class FunkyTouchExtension extends Minz_Extension {

    public function init() {
		$this->registerHook('nav_menu', [$this, 'addNavButtons']);
	}

	public function addNavButtons(): String {
		return "X";
	}
}
```

I’ve probably missed something in the documentation.

Thank you :)

**Reply by Alkarex** [reply, marked as answer]
`nav_menu` is just the closest we have right now. But we could add another hook to extend the navigation menu. PR welcome!

**Comment by 9pr19** [comment]
> `nav_menu` is just the closest we have right now. But we could add another hook to extend the navigation menu. PR welcome!

@Alkarex thank you for your help. I will have a look and hopefully I would be able to propose a PR.
